inc/saihostintf.h

```diff
@@ -461,120 +469,23 @@ typedef sai_status_t(*sai_get_hostif_trap_attribute_fn)(
         _Inout_ sai_attribute_t *attr_list);
 
 /**
- * @brief Host interface user defined trap ID table range
- */
-#define SAI_HOSTIF_USER_DEFINED_TRAP_ID_TABLE_RANGE 0x1
-
-/**
- * @brief user defined traps
- */
-typedef enum _sai_hostif_user_defined_trap_id_t
-{
-    /** Samplepacket traps */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_SAMPLEPACKET = 0x00000000,
-
-    /** ACL traps */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_ACL_MIN = 0x00000001,
-
-    /** ACL traps range max */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_ACL_MAX = SAI_HOSTIF_USER_DEFINED_TRAP_ID_ACL_MIN + SAI_HOSTIF_USER_DEFINED_TRAP_ID_TABLE_RANGE,
-
-    /** router traps */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_ROUTER_MIN = SAI_HOSTIF_USER_DEFINED_TRAP_ID_ACL_MAX + 1,
-
-    /** router traps max */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_ROUTER_MAX = SAI_HOSTIF_USER_DEFINED_TRAP_ID_ROUTER_MIN + SAI_HOSTIF_USER_DEFINED_TRAP_ID_TABLE_RANGE,
-
-    /** neighbor table traps */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_NEIGH_MIN = SAI_HOSTIF_USER_DEFINED_TRAP_ID_ROUTER_MAX+1,
-
-    /** neighbor table traps max */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_NEIGH_MAX = SAI_HOSTIF_USER_DEFINED_TRAP_ID_NEIGH_MIN + SAI_HOSTIF_USER_DEFINED_TRAP_ID_TABLE_RANGE,
-
-    /** fdb traps */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_FDB_MIN = SAI_HOSTIF_USER_DEFINED_TRAP_ID_NEIGH_MAX + 1,
-
-    /** fdb traps max */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_FDB_MAX = SAI_HOSTIF_USER_DEFINED_TRAP_ID_FDB_MIN + SAI_HOSTIF_USER_DEFINED_TRAP_ID_TABLE_RANGE,
-
-    /** Custom range base */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_CUSTOM_RANGE_BASE = 0x10000000,
-
-} sai_hostif_user_defined_trap_id_t;
-
-/**
- * @brief Host interface user defined traps attributes
+ * @brief Attribute data for SAI_HOSTIF_ATTR_TYPE
  */
-typedef enum _sai_hostif_user_defined_trap_attr_t
+typedef enum _sai_hostif_type_t
 {
-    /**
-     * @brief Start of attributes
-     */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ATTR_START,
+    /** Port-based Host Interface Type */
+    SAI_HOSTIF_TYPE_PORT,
```

Comment:
@itaibaz , please remove SAI_HOSTIF_TYPE_PORT/LAG/VLAN in the latest header, it is not in redundant.